src/pages/workspace/WorkspaceMembersPage.tsx

```diff
@@ -1,143 +1,135 @@
 import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
 import * as PolicyUtils from '@libs/PolicyUtils';
 import * as UserUtils from '@libs/UserUtils';
-import personalDetailsPropType from '@pages/personalDetailsPropType';
 import * as Policy from '@userActions/Policy';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {PersonalDetails, PersonalDetailsList, PolicyMember, PolicyMembers} from '@src/types/onyx';
+import type {Errors, Icon, PendingAction} from '@src/types/onyx/OnyxCommon';
 import SearchInputManager from './SearchInputManager';
-import {policyDefaultProps, policyPropTypes} from './withPolicy';
+import type {WithPolicyAndFullscreenLoadingProps} from './withPolicyAndFullscreenLoading';
 import withPolicyAndFullscreenLoading from './withPolicyAndFullscreenLoading';
 
-const propTypes = {
-    /** All personal details asssociated with user */
-    personalDetails: PropTypes.objectOf(personalDetailsPropType),
-
-    /** URL Route params */
-    route: PropTypes.shape({
-        /** Params from the URL path */
-        params: PropTypes.shape({
-            /** policyID passed via route: /workspace/:policyID/members */
-            policyID: PropTypes.string,
-        }),
-    }).isRequired,
-
-    /** Session info for the currently logged in user. */
-    session: PropTypes.shape({
-        /** Currently logged in user accountID */
-        accountID: PropTypes.number,
-    }),
-
-    isLoadingReportData: PropTypes.bool,
-    ...policyPropTypes,
-    ...withLocalizePropTypes,
-    ...windowDimensionsPropTypes,
-    ...withCurrentUserPersonalDetailsPropTypes,
-    network: networkPropTypes.isRequired,
+type WorkspaceMembersPageOnyxProps = {
+    /** Personal details of all users */
+    personalDetails: OnyxEntry<PersonalDetailsList>;
 };
 
-const defaultProps = {
-    personalDetails: {},
-    session: {
-        accountID: 0,
-    },
-    isLoadingReportData: true,
-    ...policyDefaultProps,
-    ...withCurrentUserPersonalDetailsDefaultProps,
+type WorkspaceMembersPageProps = WithPolicyAndFullscreenLoadingProps & WithCurrentUserPersonalDetailsProps & WorkspaceMembersPageOnyxProps;
+
+type MemberOption = {
+    keyForList: string;
+    accountID: number;
+    isSelected: boolean;
+    isDisabled: boolean;
+    text: string;
+    alternateText: string;
+    rightElement: React.ReactNode | null;
+    icons: Icon[];
+    errors?: Errors;
+    pendingAction?: PendingAction;
+    invitedSecondaryLogin?: string;
 };
 
-function WorkspaceMembersPage(props) {
+/**
+ * Inverts an object, equivalent of _.invert
+ */
+function invertObject(object: Record<string, string>): Record<string, string> {
+    const obj: Record<string, string> = {};
+    for (const key in object) {
+        if (Object.prototype.hasOwnProperty.call(object, key)) {
+            obj[object[key]] = key;
+        }
+    }
+    return obj;
```

Comment:
I think we can simplify it, for example:
```suggestion
    const invertedEntries = Object.entries(object).map(([key, value]) => [value, key]);
    return Object.fromEntries(invertedEntries);
```
Also if this one is reused, it would be good to put it in a separate place

Reply:
Great idea! I simplified it based on your suggestion. I don't think we have to make this global as `_.invert` is not used anywhere else.